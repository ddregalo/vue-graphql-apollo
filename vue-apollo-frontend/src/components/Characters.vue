<template>
  <div class="container">
    <div class="characters">
      <b-card-group deck>
        <div v-for="(character, index) in allCharacters" :key="character.name">
          <Character :role="character" :id="index"></Character>
        </div>
      </b-card-group>
    </div>
  </div>
</template>

<script>
import Character from '@/components/Character.vue';
import gql from 'graphql-tag';

export default {
  name: "characters",
  components: {
    Character
  },
  apollo: {
    allCharacters: gql`
      query {
        allCharacters {
          name
          gender
          eye_color
        }
      }
    `
  },
};
</script>

<style scoped lang="scss">
  .characters {
    padding-top: 30px;
    text-align: left;
  }
</style>
