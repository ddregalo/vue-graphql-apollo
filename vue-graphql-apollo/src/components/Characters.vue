<template>
    <div class="container">
        <div v-for="character in characters" :key="character.name">
            <Character :character="character" />
        </div>
    </div>
</template>

<script lang='ts'>
import Vue from 'vue';
import Character from './Character.vue';
export default Vue.extend({
    name: 'Characters',
    components: {
        Character,
    },
    data() {
        return {
            characters: [
                {
                    name: 'Katie',
                    height: "5'8 in.",
                    mass: '50 kg',
                    gender: 'F',
                    homeworld: 'Babeland'
                },
                {
                    name: 'Daniel',
                    height: "5'10 in.",
                    mass: '73 kg',
                    gender: 'M',
                    homeworld: 'Babeland'
                }
            ],
            msg: 'Daniel is aweaome',
        }
    },
});
</script>
