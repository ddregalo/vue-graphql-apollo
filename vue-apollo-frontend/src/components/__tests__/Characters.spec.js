import { shallowMount } from "@vue/test-utils";
import Characters from "@/components/Characters.vue";
import Vue from "vue";

Vue.config.ignoredElements = ["b-card-group"];

describe("Characters.vue", () => {
  it("renders the correct html for single charater snapshot", () => {
    const wrapper = shallowMount(Characters);
    wrapper.setData({
      allCharacters: [
        {
          name: "DarkSideOfTheMoon",
          gender: "male",
          eye_color: "brown",
        }
      ]
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
