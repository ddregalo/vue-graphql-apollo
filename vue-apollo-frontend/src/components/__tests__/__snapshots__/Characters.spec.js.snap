// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters.vue renders the correct html for single charater snapshot 1`] = `
<div
  class="container"
>
  <div
    class="characters"
  >
    <b-card-group
      deck=""
    />
  </div>
</div>
`;
